crates/wasmtime/src/runtime/vm/instance/allocator/pooling.rs

```diff
@@ -712,21 +732,28 @@ unsafe impl InstanceAllocator for PoolingInstanceAllocator {
         ty: &wasmtime_environ::Table,
         _table_index: DefinedTableIndex,
     ) -> Result<(super::TableAllocationIndex, Table)> {
-        // FIXME: see `allocate_memory` above for comments about duplication
-        // with `with_flush_and_retry`.
-        let e = match self.tables.allocate(request, ty).await {
-            Ok(result) => return Ok(result),
-            Err(e) => e,
-        };
+        self.live_tables.fetch_add(1, Ordering::AcqRel);
```

Comment:
I copied this approach from `increment_component_instance_count` but I could easily be convinced that it would be fine to just increment after successful allocation.

Reply:
Given the need to have a `Drop` to handle the cancelled-the-future case I'd be ok just putting this in the `Ok` path since that would avoid the need for `Drop` entirely